examples/plot_fmri_data_example.py

```diff
@@ -0,0 +1,173 @@
+        y = np.asarray((conditions[condition_mask] == 'cat') * 2 - 1)
+    else:
+        raise ValueError(f'Not a valid comparision: {comparision}')
+
+    groups = session_label[condition_mask]
+
+    # Loading anatomical image (back-ground image)
+    if haxby_dataset.anat[0] is None:
+        bg_img = None
+    else:
+        bg_img = mean_img(haxby_dataset.anat)
+
+    # Loading mask
+    mask_img = haxby_dataset.mask
+    masker = NiftiMasker(mask_img=mask_img, standardize=True,
+                         smoothing_fwhm=None, memory=memory)
+
+    # Computing masked data
+    fmri_masked = masker.fit_transform(fmri_filename)
+    mask = masker.mask_img_.get_fdata().astype(bool)
+
+    X = np.asarray(fmri_masked)[condition_mask, :]
+    X = StandardScaler().fit_transform(X)
+
+    return Bunch(X=X, y=y, groups=groups, bg_img=bg_img, mask=mask,
+                 masker=masker)
+
+
+#############################################################################
+# Gathering and preprocessing Haxby dataset for a given subject
+# -------------------------------------------------------------
+list_subject = [1, 2, 3, 4, 5, 6]
+subject = list_subject[1]
+data = preprocess_haxby(subject=subject)
+X, y, groups, mask = data.X, data.y, data.groups, data.mask
+
+#############################################################################
+# Initializing FeatureAgglomeration object needed to perform the clustering
+# -------------------------------------------------------------------------
+# For fMRI data n_clusters = 500 is generally a good default choice.
+n_clusters = 500
+# Deriving connectivity.
+shape = mask.shape
+n_x, n_y, n_z = shape[0], shape[1], shape[2]
+connectivity = image.grid_to_graph(n_x=n_x, n_y=n_y, n_z=n_z, mask=mask)
+# Initializing FeatureAgglomeration object
+ward = FeatureAgglomeration(n_clusters=n_clusters, connectivity=connectivity)
+
+#############################################################################
+# Making the inference with several algorithms
+# --------------------------------------------
+# Setting theoretical FWER target
+n_samples, n_features = X.shape
+fwer_target = 0.1
+# No need of correction for permutation tests
+zscore_threshold_corr = zscore_from_pval((fwer_target / 2))
+# Other methods need to be corrected
+correction = 1. / n_features
+zscore_threshold_no_clust = zscore_from_pval((fwer_target / 2) * correction)
+correction_clust = 1. / n_clusters
+zscore_threshold_clust = zscore_from_pval((fwer_target / 2) * correction_clust)
+
+# Recovering the support with SVR decoder thresholded parametrically
+# We computed the regularization parameter by CV (C = 0.1)
+beta_hat, scale = standardized_svr(X, y, Cs=[0.1])
+pval, _, one_minus_pval, _ = pval_from_scale(beta_hat, scale)
+zscore_std_svr = zscore_from_pval(pval, one_minus_pval)
+
+# Recovering the support with SVR decoder thresholded by permutation test
+# This inference takes around 15 minutes.
```

Comment:
Oops. Isn't it possible to rely on another estimator for the sake of time ?

Reply:
You would like to a Ridge decoder thresholded by permutation test for example ?

Reply:
Yes, with a word to explain the reason: computation time.